### PR TITLE
copied files from original folder

### PR DESCRIPTION
**Pull request by muudi10**

### CODE REVIEW

File: src/kata1.fizzBuzz.js [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+      return "Fizz";
+    }
+  
+    if (number % 5 === 0) {
+      return "Buzz";
+    }
+  
+    return number;
+  };
+  console.log(fizzBuzz(30));
```
Comment (ruimiguelcorreia) [inline src/kata1.fizzBuzz.js:15]: As a good practice for the future, you should clean your code from console.log statements before creating that PR. It's not a big deal, but your peers will tell you to remove that.

File: src/kata2.booleanToWord.js [hunk 2]
```diff
@@ -1,3 +1,11 @@
-const booleanToWord = boolean => {};
+const booleanToWord = boolean => {
+    if (boolean === true) {
+    return "yes";
+    }
+    else {
+    return "no";
+    }
```
Comment (ruimiguelcorreia) [inline src/kata2.booleanToWord.js:7]: You could have used Boolean(boolean) here and a ternary operator, if you want a cleaner approach.
Yours is fine, anyway.
Careful with indentation.

File: src/kata4.humanCatDogYears.js [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+      if (i === 2) {
+        dogYears += 9;
+        catYears += 9;
+      }
+      if (i > 2) {
+        dogYears += 5;
+        catYears += 4;
+      }
+    }
+
```
Comment (ruimiguelcorreia) [inline src/kata4.humanCatDogYears.js:21]: Interesting solution!

File: src/kata4.humanCatDogYears.js [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+    } if (sequence.length >2){
+  
+      catAge = 24 +((maxValue-1)*4)
+      DogYears = 24 + (maxValue - 1) * 5;
+      yearsArray.push(catAge);
+      yearsArray.push(DogYears);
+    }
+  
+    
+     return yearsArray;
```
Comment (ruimiguelcorreia) [inline src/kata4.humanCatDogYears.js:62]: It is an overkill indeed, but valid.

File: src/kata6.joinNames.js [hunk 5]
```diff
@@ -1,3 +1,33 @@
...
+
+if (namesList.length > 2 ) {
+lastName = namesList.pop()
+
+var n = 1;
+joinednames =
+  namesList.splice(namesList - n).join(", ") + " &" + " " + lastName;
+ 
+return joinednames;
+}
```
Comment (ruimiguelcorreia) [inline src/kata6.joinNames.js:26]: I will challenge you to solve this with reduce! It's much cleaner

File: test/kata4.humanCatDogYears.test.js [hunk 6]
```diff
@@ -1,3 +1,10 @@
 const { humanCatDogYears } = require("../src");
 
+describe("convertts human years to cat and dog years", () => {
+ it.only('Converts Human years to cat years and dog years vaguely ',()=>{
+ expect(humanCatDogYears(10)).toEqual([10,56,64])
+ expect(humanCatDogYears(5)).toEqual([5,36,39])
+  });
```
Comment (ruimiguelcorreia) [inline test/kata4.humanCatDogYears.test.js:7]: I would also test 1 and 2, since those are the 2 variations.

File: test/kata5.reachDestination.test.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
 const { reachDestination } = require("../src");
 
 describe("reachDestination", () => {
-  test("returns string with estimated time of arrival", () => {});
+  test("returns string with estimated time of arrival", () => {
+ 
+    expect(reachDestination(60,60)).toEqual('1hr')
+    expect(reachDestination(120, 60)).toEqual("2hr");
```
Comment (ruimiguelcorreia) [inline test/kata5.reachDestination.test.js:7]: What if we have a division that doesn't return an integer? Like 135/60, which equals 2.25?
What would be the time to reach the destination?
How would you handle the minutes?

File: test/kata6.joinNames.test.js [hunk 8]
```diff
@@ -1,5 +1,9 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  it.only("returns string of names, seperated by commas and an ampersand", () => {
+expect(joinNames([{ name: "Bart" }, { name: "Ali" }, { name: "Omar" }, {name: 'Zahra'}])).toEqual('Bart, Ali, Omar & Zahra')
+
```
Comment (ruimiguelcorreia) [inline test/kata6.joinNames.test.js:6]: You should test a case where only 2 names are provided, or 1, since those are the odd ones.

**Review by ruimiguelcorreia — CHANGES_REQUESTED** [review]
Overall very good work.
There's some refactoring to do, and if you add some extra tests, you would probably have to tweak your logic.

You should aim to write your tests in a way that looks at edge cases instead of certain values, which improves the robustness of your function!

File: test/kata6.joinNames.test.js (hunk 8 above)
Comment (muudi10) [reply test/kata6.joinNames.test.js:6]: I have actually forgot to test will names, but now have done so and the solution works

File: src/kata6.joinNames.js (hunk 5 above)
Comment (muudi10) [reply src/kata6.joinNames.js:26]: I will attempt as fresh solution under this code.
